lib/go-backoff/backoff.go

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package backoff
+
+/*
+ * Used to return an exponentially increasing time.Duration value that may be used
+ * to sleep between retries.
+ */
+
+import (
+	"github.com/apache/trafficcontrol/lib/go-log"
+	"math"
+	"math/rand"
+	"time"
+)
+
+// default minimum milliseconds is 100 msec
+const DefaultMinMs = time.Duration(100) * time.Millisecond
+
+// default maximum millseconds is 60000 (1m 0s)
+const DefaultMaxMs = time.Duration(60000) * time.Millisecond
+
+const DefaultFactor = float64(2)
```

Comment:
Nitpick: the cast is unnecessary, if you make the literal a float, it will default to type `float64`: `const DefaultFactor = 2.0`